docs/reference/analysis/analyzers.asciidoc

```diff
@@ -1,71 +1,148 @@
-referenced from mapping definitions or certain APIs. When none are
-defined, defaults are used. There is an option to define which analyzers
-will be used by default when none can be derived.
+* The `default` analyzer is used for both indexing and searching. 
+* The `default_index` analyzer is used only when indexing.
+* The `default_search` analyzer is be used only when searching.
 
-The `default` logical name allows one to configure an analyzer that will
-be used both for indexing and for searching APIs. The `default_index`
-logical name can be used to configure a default analyzer that will be
-used just when indexing, and the `default_search` can be used to
-configure a default analyzer that will be used just when searching.
+In most cases, you want to use the same analyzer for both indexing and searching,
+so setting the `default` analyzer is recommended. For example:
+
+[source,js]
+--------------------------------------------------
+include::../snippets/analysis/configure-default-analyzer.json[]
+--------------------------------------------------
+// SENSE: analysis/configure-default-analyzer.json
+
+If no default is configured and you don't specify an analyzer when configuring mappings
+or searching, Elasticsearch uses the <<analysis-standard-analyzer, `standard`>> analyzer.
 
 [float]
 [[aliasing-analyzers]]
 === Aliasing Analyzers
 
-Analyzers can be aliased to have several registered lookup names
-associated with them. For example, the following will allow 
-the `standard` analyzer to also be referenced with `alias1`
-and `alias2` values.
+You can define one or more aliases for an analyzer so you can reference it 
+by a custom name. 
 
+To configure an alias, specify the `alias` attribute in the `analyzer` configuration.
+For example, the following configuration enables you to reference the `standard` 
+analyzer configured with the stopwords `a` and `an` by either 
+`alias1` or `alias2`. 
 
 [source,js]
 --------------------------------------------------
-index :
-  analysis :
-    analyzer :
-      standard :
-        alias: [alias1, alias2]
-        type : standard
-        stopwords : [test1, test2, test3]
+include::../snippets/analysis/configure-analyzer-alias.json[]
 --------------------------------------------------
+// SENSE: analysis/configure-analyzer-alias-analyzer.json
 
-Below is a list of the built in analyzers.
-
-include::analyzers/standard-analyzer.asciidoc[]
+// ^^^ Why would you want to define multiple aliases?
 
-include::analyzers/simple-analyzer.asciidoc[]
+[float]
+[[backwards-compatibility]]
+=== Backward Compatibility
 
-include::analyzers/whitespace-analyzer.asciidoc[]
+All analyzers, tokenizers, and token filters can be configured with a
+`version` parameter to control which Lucene version behavior they should
+use. Possible values are: `3.0` - `3.6`, `4.0` - `4.3` (the highest
+version number is the default option).
 
-include::analyzers/stop-analyzer.asciidoc[]
+include::analyzers/custom-analyzer.asciidoc[]
```

Comment:
As a compromise, have the ToC alphabetical, but on the summary page list the analyzers (with a short description) in the order that they should be read?